pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -748,6 +751,19 @@ func (r *RepoSyncReconciler) validateNamespaceName(namespaceName string) error {
 	return nil
 }
 
+func (r *RepoSyncReconciler) validateCACertSecret(ctx context.Context, repoSync *v1beta1.RepoSync) error {
```

Comment:
Need unit tests for both reposync_controller and rootsync_controller. This should include test cases with valid and invalid CACertSecret configurations